README.md

```diff
@@ -1,39 +1,96 @@
+self-signed [certificates](./certs) that can be used for testing purposes.
+
+### Docker CLI
 
 ```bash
-docker compose up
+docker run --name hello -p 4062:4062/tcp -p 4062:4062/udp -v /path/to/certificates/:/certs icerpc/hello
 ```
 
-Once running, the server will be available at `icerpc://localhost` using the `TCP` and `QUIC` transports on the default
-port (4062).
+Optional environment variables can be used to configure the server:
+
+```bash
+docker run \
+  --name hello \
+  -p 4062:4062/tcp -p 4062:4062/udp \
+  -v /path/to/certificates/:/certs \
+  -e LOG_LEVEL= \
+  -e SERVER_CERT= \
+  -e SERVER_KEY= \
+  -e USE_TLS_WITH_TCP=true icerpc/hello
+```
+
+### Docker Compose
+
+```yml
+services:
+  hello:
+    image: icerpc/hello
+    ports:
+      - "4062:4062/tcp"
+      - "4062:4062/udp"
+    environment:
+      - LOG_LEVEL= #optional
+      - SERVER_CERT= #optional
+      - SERVER_KEY= #optional
+      - USE_TLS_WITH_TCP= #optional
+    volumes:
+      - /path/to/certificates/:/certs
 
-## Configuration
+```
+
+### Configuration
 
 The server can be configured through several environment variables:
 
-- `LOG_LEVEL` - The [log level](https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-level). Default: `Debug`
-- `SERVER_CERT` - Path to the server certificate. Default: `/certs/server_cert.pem`
-- `SERVER_KEY` - Path to the server private key. Default: `/certs/server_key.pem`
-- `USE_TLS_WITH_TCP` - Configure the usage of TLS with the TCP transport. Default: `true`
+| Environment Variable | Description                                                                                                   | Default Value            |
+| -------------------- | ------------------------------------------------------------------------------------------------------------- | ------------------------ |
+| `LOG_LEVEL`          | The [log level](https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-level) | `Debug`                  |
+| `SERVER_CERT`        | Path to the server certificate                                                                                | `/certs/server_cert.pem` |
+| `SERVER_KEY`         | Path to the server private key                                                                                | `/certs/server_key.pem`  |
+| `USE_TLS_WITH_TCP`   | Configure the usage of TLS with the TCP transport                                                             | `true`                   |
+
+Certificates are **required** to run the server as the QUIC transport requires TLS.
+
+## Building with Docker Compose
+
+This repository contains a [`Dockerfile`](./Dockerfile) that can be used to build the server as well as a
+[`compose.yml`](./compose.yml) file that can be used to build and run the server locally
+using Docker Compose.
+
+This is the recommended way to build and run the server as QUIC support in .NET is not available on all systems.
```

Comment:
I would remove this sentence. The ability to run QUIC on any system is not what this container is about.